src/CoreWCF.Primitives/src/CoreWCF/Channels/SecurityBindingElement.cs

```diff
@@ -298,52 +299,172 @@ private static BindingContext CreateIssuerBindingContextForNegotiation(BindingCo
+        {
+            SymmetricSecurityBindingElement result = new SymmetricSecurityBindingElement(
+                new SslSecurityTokenParameters(requireClientCertificate, requireCancellation));
+            return result;
+        }
+
+        public static SymmetricSecurityBindingElement CreateIssuedTokenBindingElement(IssuedSecurityTokenParameters issuedTokenParameters)
+        {
+            if (issuedTokenParameters == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(issuedTokenParameters));
+            if (issuedTokenParameters.KeyType != SecurityKeyType.SymmetricKey)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgument(SR.Format(SR.IssuedTokenAuthenticationModeRequiresSymmetricIssuedKey));
+            SymmetricSecurityBindingElement result = new SymmetricSecurityBindingElement(issuedTokenParameters);
+            return result;
+        }
+
+        // If any changes are made to this method, please make sure that they are
+        // reflected in the corresponding IsIssuedTokenForCertificateBinding() method.
+        public static SymmetricSecurityBindingElement CreateIssuedTokenForCertificateBindingElement(IssuedSecurityTokenParameters issuedTokenParameters)
+        {
+            if (issuedTokenParameters == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(issuedTokenParameters));
+
+            SymmetricSecurityBindingElement result = new SymmetricSecurityBindingElement(
+                new X509SecurityTokenParameters(
+                    X509KeyIdentifierClauseType.Thumbprint,
+                    SecurityTokenInclusionMode.Never));
+            if (issuedTokenParameters.KeyType == SecurityKeyType.BearerKey)
             {
-                SetIssuerBindingContextIfRequired(supportingParameters.Signed[i], issuerBindingContext);
+                result.EndpointSupportingTokenParameters.SignedEncrypted.Add(issuedTokenParameters);
+                result.MessageSecurityVersion = MessageSecurityVersion.WSSXDefault;
             }
-            for (int i = 0; i < supportingParameters.SignedEncrypted.Count; ++i)
+            else
             {
-                SetIssuerBindingContextIfRequired(supportingParameters.SignedEncrypted[i], issuerBindingContext);
+                result.EndpointSupportingTokenParameters.Endorsing.Add(issuedTokenParameters);
+                result.MessageSecurityVersion = MessageSecurityVersion.Default;
+            }
+            result.RequireSignatureConfirmation = true;
+            return result;
+        }
+
+        // this method reverses CreateMutualCertificateBindingElement() logic
+        internal static bool IsIssuedTokenForCertificateBinding(SecurityBindingElement sbe, out IssuedSecurityTokenParameters issuedTokenParameters)
+        {
```

Comment:
Remove all the IsXXXXX methods as they are client usage only and used for interpreting an imported WSDL

Reply:
removed all except IsUnderlyingDispatcherDuplex (this has nothing to do with WSDL)